### PR TITLE
VCR: Support verifying VPs

### PR DESCRIPTION
**Pull request by reinkrul**
Based on https://github.com/nuts-foundation/nuts-node/pull/844

### CODE REVIEW

File: docs/_static/vcr/v2.yaml [hunk 1]
```diff
@@ -210,6 +210,38 @@ paths:
...
+      description: |
+        Verifies a Verifiable Presentation. It checks:
+        * Signature of the verifiable presentation and the verifiable credentials
+        * Expiration
+        * Revocation status
+        * If the issuers of the verifiable credentials are trusted
+        * If the issuers of the verifiable credentials were not deactivated at time of issuing
+
+        error returns:
+        * 400 - One or more of the given parameters are invalid
```
Comment (woutslakhorst) [inline docs/_static/vcr/v2.yaml:225]: I would expect one of the above rules to fail since the call has no params.

File: vcr/api/v2/api.go [hunk 2]
```diff
@@ -229,3 +231,34 @@ func (w *Wrapper) CreateVP(ctx echo.Context) error {
 	}
 	return ctx.JSON(http.StatusOK, vp)
 }
+
+// VerifyVP handles API request to verify a Verifiable Presentation.
+func (w *Wrapper) VerifyVP(ctx echo.Context) error {
```
Comment (woutslakhorst) [inline vcr/api/v2/api.go:236]: The number of checks do not represent the number of 400 errors that are returned by this code. It seems some are returned as 500s?

File: vcr/api/v2/api_test.go [hunk 3]
```diff
@@ -540,6 +541,63 @@ func TestWrapper_CreateVP(t *testing.T) {
...
+			return nil
+		})
+		testContext.mockVerifier.EXPECT().VerifyVP(vp, false, nil).Return(vp.VerifiableCredential, nil)
+		testContext.echo.EXPECT().JSON(http.StatusOK, vp.VerifiableCredential)
+
+		err := testContext.client.VerifyVP(testContext.echo)
+
+		assert.NoError(t, err)
+	})
+	t.Run("error - verification failed", func(t *testing.T) {
```
Comment (woutslakhorst) [inline vcr/api/v2/api_test.go:585]: Missing tests on other error conditions as stated in OAS

File: vcr/api/v2/api.go (hunk 2 above)
Comment (woutslakhorst) [reply vcr/api/v2/api.go:236]: This could simply be fixed by mapping the validationError to a status code.

**Review by woutslakhorst — CHANGES_REQUESTED** [review]
Also some linter errors

File: docs/_static/vcr/v2.yaml (hunk 1 above)
Comment (reinkrul) [reply docs/_static/vcr/v2.yaml:225]: I'll change the 400 description to parameters or VP.

Also, 400 is returned when the VP itself can't be verified (e,g. invalid signature, not trusted), which is ambiguous. So i'll change that to 422

File: vcr/api/v2/api.go (hunk 2 above)
Comment (reinkrul) [reply vcr/api/v2/api.go:236]: 400 is returned when `Bind()` fails (see `github.com/labstack/echo/v4@v4.6.3/bind.go:73`)

File: vcr/api/v2/api_test.go (hunk 3 above)
Comment (reinkrul) [reply vcr/api/v2/api_test.go]: Added tests for failed VP validation, 400 and 500 are returned by Echo

**Review by stevenvegt — APPROVED** [review]
LGTM

**Comment by reinkrul** [comment]
> LGTM

Thanks, but wasn't even finished yet ;-)

**Review by woutslakhorst — APPROVED** [review]
only question would be the 422 vs 200. A valid response with an unverified VP, would that be a 200 or 422?

A use may try the call again, the outcome would be the same but it could. 4xx return calls say that the same call may not be done again.

**Pull request merged**